### PR TITLE
feat: refactor vector store to wrap async class

### PR DESCRIPTION
**Pull request by averikitsch**
Current diff is against async-only branch

### CODE REVIEW

File: src/langchain_google_cloud_sql_pg/vectorstore.py [hunk 1]
```diff
@@ -357,35 +224,31 @@ def add_documents(
...
         """Delete records from the table."""
-        return self.engine._run_as_sync(self.adelete(ids, **kwargs))
+        return self._engine._run_as_sync(self._vs.adelete(ids, **kwargs))
 
     @classmethod
     async def afrom_texts(  # type: ignore[override]
         cls: Type[PostgresVectorStore],
         texts: List[str],
-        embedding: Embeddings,
+        embedding_service: Embeddings,
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/vectorstore.py:251]: breaking change?

File: src/langchain_google_cloud_sql_pg/vectorstore.py [hunk 2]
```diff
@@ -465,27 +343,32 @@ async def afrom_documents(  # type: ignore[override]
...
+            k,
+            fetch_k,
+            lambda_mult,
+            index_query_options,
         )
-        texts = [doc.page_content for doc in documents]
-        metadatas = [doc.metadata for doc in documents]
-        await vs.aadd_texts(texts, metadatas=metadatas, ids=ids, **kwargs)
+        async_vs = await engine._run_as_async(coro)
+        vs = cls(cls.__create_key, engine, async_vs)
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/vectorstore.py:363]: nit: should use create method

File: src/langchain_google_cloud_sql_pg/vectorstore.py [hunk 3]
```diff
@@ -547,7 +438,11 @@ def from_documents(  # type: ignore[override]
         ignore_metadata_columns: Optional[List[str]] = None,
         id_column: str = "langchain_id",
         metadata_json_column: str = "langchain_metadata",
-        **kwargs: Any,
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        k: int = 4,
+        fetch_k: int = 20,
+        lambda_mult: float = 0.5,
+        index_query_options: Optional[QueryOptions] = None,
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/vectorstore.py:445]: Need to update the docstring

File: src/langchain_google_cloud_sql_pg/vectorstore.py [hunk 4]
```diff
@@ -568,78 +463,60 @@ def from_documents(  # type: ignore[override]
...
         # vectorstore constructor
-        if self.distance_strategy == DistanceStrategy.COSINE_DISTANCE:
+        if self._vs.distance_strategy == DistanceStrategy.COSINE_DISTANCE:
             return self._cosine_relevance_score_fn
-        if self.distance_strategy == DistanceStrategy.INNER_PRODUCT:
+        if self._vs.distance_strategy == DistanceStrategy.INNER_PRODUCT:
             return self._max_inner_product_relevance_score_fn
-        elif self.distance_strategy == DistanceStrategy.EUCLIDEAN:
+        elif self._vs.distance_strategy == DistanceStrategy.EUCLIDEAN:
             return self._euclidean_relevance_score_fn
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/vectorstore.py:520]: Should this be implemented in the underlying vectorstore?

Comment (kurtisvg) [reply src/langchain_google_cloud_sql_pg/vectorstore.py:520]: follow up: probbaly not, looks like it's relying on inherit values? should confirm

File: src/langchain_google_cloud_sql_pg/vectorstore.py [hunk 5]
```diff
@@ -895,121 +698,53 @@ async def aapply_vector_index(
...
+
+    def drop_vector_index(
+        self,
+        index_name: Optional[str] = None,
+    ) -> None:
+        """Drop the vector index."""
+        return self._engine._run_as_sync(self._vs.adrop_vector_index(index_name))
 
-    async def is_valid_index(
+    async def ais_valid_index(
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/vectorstore.py:738]: this is breaking change, make sure we flag in commit message

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: src/langchain_google_cloud_sql_pg/vectorstore.py (hunk 4 above)
Comment (averikitsch) [reply src/langchain_google_cloud_sql_pg/vectorstore.py]: I added a comment that this is used for similarity_search_with_relevance_scores() which is predefined by the interface and calls other defined mehtods. The tests are passing.

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**